quakesaver_client/fdsnws.py

```diff
@@ -30,37 +27,34 @@ class FDSNWSDataselectQuery(BaseModel):
     station: constr(max_length=5) = "*"
     location: constr(max_length=2) = "*"
     channel: constr(max_length=3) = "*"
-    quality: DataQuality = "B"
 
     longestonly: bool = True
-    minimumlength: PositiveFloat = 0.0
 
     format: DataFormat = "miniseed"
-    nodata: NoData = 204
 
 
 def dataselect(
     uri: str,
     params: FDSNWSDataselectQuery,
-    location_to_store: Path | str = None,
-) -> Path:
-    """Request FDSN waveform data of the sensor and stores in as a miniseed file."""
+    buffer: BinaryIO,
+) -> str:
+    """Request FDSN waveform data of the sensor and stores in as a MiniSEED file."""
     logging.debug("requesting waveform data for sensor %s.", uri)
-    location_to_store = assure_output_path(location_to_store)
     response = requests.get(
-        url=f"{uri}/fdsnws/dataselect/1/query",
-        params=params.json(),
+        url=f"{uri}/fdsnws/dataselect/1/query", params=params.dict()
     )
 
-    response.raise_for_status()
-    if response.status_code in NoData.__args__:
+    if response.status_code in get_args(NoData):
         raise NoDataError(response.text)
+    response.raise_for_status()
+
+    try:
+        filename = response.headers.get("Content-Disposition").split("=")[1].strip('"')
+    except (KeyError, IndexError):
+        filename = "qssensor-data.mseed"
```

Comment:
It is used downstream...

Reply:
Ah, I misunderstood the feature request then. I understood that it would be favorable having a way to pass in a file handle to which the file is writen, e.g. to avoid writing to disk.